### PR TITLE
DragonFly BSD supports `sched_getaffinity()` and `sched_setaffinity()`

### PR DESCRIPTION
**Pull request by rtzoeller**
Move `CpuSet`, `sched_getaffinity()`, and `sched_setaffinity()` from the `sched_linux_like` namespace to `sched_affinity`. Declare support for DragonFly BSD.

`test_sched_affinity()` passes locally on DragonFly BSD.

### CODE REVIEW

File: Cargo.toml
Comment (rtzoeller) [inline Cargo.toml:28]: I'm fine waiting until a new libc release is cut for this to be merged. No urgency on my end.

File: src/sched.rs
Comment (rtzoeller) [inline src/sched.rs:147]: This is technically a breaking change if anyone was importing these functions from `nix::sched::sched_linux_like`, but I'm not sure why they would since `nix::sched` exports them.

**Review by asomers — DISMISSED** [review]
LGTM, as long as you fix the CHANGELOG.

**Comment by rtzoeller** [comment]
bors retry

**Pull request merged**
